aten/src/ATen/MemoryOverlap.h

```diff
@@ -0,0 +1,13 @@
+#pragma once
+
+#include <ATen/ATen.h>
+
+namespace at {
+
+enum MemOverlap { kNo, kYes, kTooHard };
```

Comment:
Make this an enum class. Also, the prevailing naming convention for enum members is all-caps, not k-prefixed (which is for non-enum constants. For example kCPU = DeviceType::CPU)

Reply:
It's pretty clear to me what this says, but for others, could we add a comment for this? Also mention that you need to update the Python test if you renumber this.